submissions/antonkottans/tjsw-post-oop/index.js

```diff
@@ -0,0 +1,140 @@
+class Creature {
+    constructor({ species, name, gender, saying, friends }) {
+        this.name = name;
+        this.gender = gender;
+        this.friends = friends;
+    }
+    returnFullInfo() {
+        const keys = [
+            "species",
+            "name",
+            "gender",
+            "legs",
+            "hands",
+            "paws",
+            "tail",
+            "saying",
+            "friends",
```

Comment:
Base class shouldn't know anything about design of its children classes. This breaks SOLID principles.